### PR TITLE
Unit Tests for: repo_test.go, task_test.go

### PR DESCRIPTION
**Pull request by oliveiradan**
@bradrydzewski here it is the new PR you requested. 
;
doliveira@doliveira-opensuse-dev10:~/go/src/github.com/drone/drone> clear; date; make test
Mon Apr 20 15:19:14 MDT 2015
go vet ./...
go test -cover -short ./...
?       github.com/drone/drone  [no test files]
?       github.com/drone/drone/builder  [no test files]
?       github.com/drone/drone/builder/docker   [no test files]
?       github.com/drone/drone/builder/pool     [no test files]
?       github.com/drone/drone/common   [no test files]
?       github.com/drone/drone/common/ccmenu    [no test files]
?       github.com/drone/drone/common/httputil  [no test files]
?       github.com/drone/drone/common/oauth2    [no test files]
?       github.com/drone/drone/common/sshutil   [no test files]
?       github.com/drone/drone/datastore        [no test files]
ok      github.com/drone/drone/datastore/bolt   10.966s coverage: 72.6% of statements
?       github.com/drone/drone/datastore/rpc    [no test files]
?       github.com/drone/drone/eventbus [no test files]
ok      github.com/drone/drone/parser   0.002s  coverage: 67.5% of statements
ok      github.com/drone/drone/parser/inject    0.002s  coverage: 100.0% of statements
ok      github.com/drone/drone/parser/matrix    0.002s  coverage: 68.6% of statements
?       github.com/drone/drone/remote   [no test files]
?       github.com/drone/drone/remote/client    [no test files]
?       github.com/drone/drone/remote/github    [no test files]
?       github.com/drone/drone/server   [no test files]
?       github.com/drone/drone/server/session   [no test files]
?       github.com/drone/drone/settings [no test files]

### CODE REVIEW

File: datastore/bolt/repo_test.go [hunk 1]
```diff
@@ -1,24 +1,138 @@
...
+		})
+
+		// we test again with same repo/user already existing
+		// to see if it will return "ErrConflict"
+		g.It("Should set SetRepoNotExists", func() {
+			err := db.SetRepoNotExists(&common.User{Login: testUser}, &common.Repo{FullName: testRepo})
+			g.Assert(err).Equal(nil)
+			// We should get ErrConflict now, trying to add the same repo again.
+			err_ := db.SetRepoNotExists(&common.User{Login: testUser}, &common.Repo{FullName: testRepo})
+			g.Assert(err_ == nil).IsFalse() // we should get (ErrConflict)
```
Comment (bradrydzewski) [inline datastore/bolt/repo_test.go]: maybe check `g.Assert(err).Equal(ErrConflict)`?

File: datastore/bolt/repo_test.go [hunk 2]
```diff
@@ -1,24 +1,138 @@
...
+		// test and destroy afterwards.
+		g.BeforeEach(func() {
+			db = Must("/tmp/drone.test.db")
+		})
+		g.AfterEach(func() {
+			os.Remove(db.Path())
+		})
+
+		g.It("Should set Repo", func() {
+			//err := db.SetRepoNotExists(&common.User{Name: testUser}, &common.Repo{Name: testRepo})
```
Comment (bradrydzewski) [inline datastore/bolt/repo_test.go]: let's remove some of these comments from the code

**Comment by bradrydzewski** [comment]
great job. just some minor cleanup (like removing some of the commented out code) and we'll get it merged!

**Comment by oliveiradan** [comment]
@bradrydzewski , on the (ErrConflict), when I tried to use it, I got an error saying it wasn't defined and that's why I commented it out.

**Comment by bradrydzewski** [comment]
maybe try `ErrKeyExists`?

**Comment by oliveiradan** [comment]
Yep, ErrKeyExists was the correct one, once we are going thru the insert(). Thanks.

**Comment by bradrydzewski** [comment]
great work!  thanks for all the effort ... I know you are new to Go and I really appreciate you sticking with this PR until ready

**Comment by oliveiradan** [comment]
Thank you for your helping and coaching!

**Pull request merged**